### PR TITLE
Improve highlighting of Liquid `for .. in` tags

### LINKED ISSUE #1194: The keyword endfor is not properly matched in the Liquid lexer
**Name of the lexer**
Liquid

**Online example**
[rouge.jneen.net](http://rouge.jneen.net) is currently not accepting new pastes at the moment.

**Code sample**
A sample of the code that produces the bug.

```liquid
{% if x %}
Do something.
{% endif %}

{% for x in xs %}
Do something.
{% endfor %}
```

**Additional context**
This was discussed in PR #1192 (see [here](https://github.com/rouge-ruby/rouge/pull/1192#discussion_r294530920)).

### PR DESCRIPTION
**Pull request by ashmaroli**
Resolves #1194

- `for` is a tag used for [iterating](https://shopify.github.io/liquid/tags/iteration/) through a given collection.

### CODE REVIEW

File: lib/rouge/lexers/liquid.rb [hunk 1]
```diff
@@ -88,6 +88,38 @@ class Liquid < RegexLexer
...
+                        else
+                          Name::Variable
+                        end
+          token token_class, m[7]
+
+          token Text::Whitespace, m[8]
+          token Name::Attribute,  m[9]
+          push :tag_markup
+        end
+
```
Comment (pyrmont) [inline lib/rouge/lexers/liquid.rb]: This is impressive but is it necessary? We already have a rule that matches an arbitrary `end<tag>` tag. Why not just get rid of the existing `end(case|unless…)` regex?

Comment (ashmaroli) [reply lib/rouge/lexers/liquid.rb]: By "this", are you referring to a particular line or `rule` or the entire PR? Sorry, I'm not able to place a finger exactly.
The focus of this PR is not the closing `end(.*)`.

It's on highlighting the "parameter" of the `for` tag, the keyword `in` and the collection reference.

Comment (ashmaroli) [reply lib/rouge/lexers/liquid.rb]: Note: The regex is taken directly from Liquid's source code.

Comment (pyrmont) [reply lib/rouge/lexers/liquid.rb]: The entire rule itself is impressive in its scope and coverage. I'm just a little wary about rules that make the lexer more of a parser. I'll have a bit of a play when I'm back at home.

Comment (pyrmont) [reply lib/rouge/lexers/liquid.rb]: Sorry, took longer for me to get around to this than I originally thought. Looks good to me.

**Pull request merged**